Fix #37

nick-iris opened the pull request:
This change allows for multiple data directories to be defined.

For the Web Services, this configuration is defined in the config files.

I think in the future, experiments should be added to a testing directory that is only accessible by the PH5 Beta Web Services first, before being moved to the master archive. This way we can assure that we don't unintentionally break the production Web Services if a incomplete experiment is added.

nick-iris commented:
@derick-hess Having both a _ph5path_ and _basepath_ seems unnecessary. Do you think we could remove the _ph5path_ argument and just initialize `paths` to _basepath_?

i.e. instead of:
`paths = self.args.get('ph5path')`
https://github.com/PIC-IRIS/PH5/blob/ee07138632059b6d0810349812a3507008580bf4/kitchen/ph5tostationxml.py#L692
have:
`paths = self.args.get('basepath')`

derick-hess commented:
Yeah I think that will work, but maybe use the ph5paths argument as the basepath argument so the command line still works properly

So you should still be able to run the command:
ph5tostationxml -n master -p /storage/Geogirls -o test.xml 

so it matches all the other programs.

nick-iris commented:
Thanks Derick, good suggestion. I removed the _basepath_ argument and changed the _ph5path_ argument to also perform a os.walk.  I now think this could be merged if you are happy with the changes.

derick-hess commented:
Thanks. I'll take a look at it now.

nick-iris commented:
I made a directory on the pic-ph5 server called /hdf5-data2/PH5_Experiments/pn4-beta/. Once the production Web Service is released, experiments should be added here for testing before being added to the master /hdf5-data2/PH5_Experiments/pn4/ archive.

nick-iris commented on kitchen/ph5tostationxml.py:
```diff
@@ -401,7 +401,7 @@ def create_obs_channel(self, station_list, deployment, cha_code, loc_code,
             Das_t = ph5API.filter_das_t(self.ph5.Das_t[das]['rows'],
                                         station_list[deployment][0]['channel_number_i'])
-            Receiver_t=self.ph5.get_receiver_t (Das_t[0], by_n_i=True)
+            Receiver_t=self.ph5.get_receiver_t (Das_t, by_n_i=True)
```
Was this what was causing the issue with experiment 16-013?

derick-hess replied:
It might have been part of it. I'm still working on that.  The full fix for it may be to give a start and end time when reading the das table in get_response_inv as well. I'm testing this now and will let you know. I will also need to fix PH5API so that 'rows' always gets set. 16-013 is putting it in a strange state and I'll let you know when i figure out why.

nick-iris replied:
I'm not sure if this is right. I just made this change and now get and error.

http://service.iris.edu/ph5beta/station/1/query?network=8A&format=TEXT&level=CHANNEL
returns:
Error 500: list indices must be integers, not str

derick-hess replied:
hmm that's very strange. I get that same error on my machine when the [0] isn't there. let's change it back for now since the [0] works fine in web services and I'll figure out what is goig on

nick-iris replied:
Okay I'm going to change the PH5 Beta code back too.

nick-iris commented:
Okay thanks for the review. Hold off on merging until we confirm that the https://github.com/PIC-IRIS/PH5/pull/38/commits/a44668ac9255522b93d7c324be462c00a139791d change was correct.

derick-hess commented:
Okay my bad... it was fine the way it was before my change. For some reason my build was using an out of date PH5API.  Cleaned the directory and cloned it again and it worked fine

nick-iris commented:
No problem. I'll reset the head to the previous commit and force push.

nick-iris commented:
Is it okay to merge now?

derick-hess commented:
Yeah it's okay to merge
